pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -246,57 +233,39 @@ func PodMatchNodeSelector(pod *v1.Pod, meta Metadata, nodeInfo *schedulernodeinf
 	return false, []PredicateFailureReason{ErrNodeSelectorNotMatch}, nil
 }
 
-// PodFitsHost checks if a pod spec node name matches the current node.
-func PodFitsHost(pod *v1.Pod, meta Metadata, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
-	if len(pod.Spec.NodeName) == 0 {
-		return true, nil, nil
-	}
+// podFitsHost checks if a pod spec node name matches the current node.
+func podFitsHost(pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
 	node := nodeInfo.Node()
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if pod.Spec.NodeName == node.Name {
+	if len(pod.Spec.NodeName) == 0 || pod.Spec.NodeName == node.Name {
 		return true, nil, nil
 	}
 	return false, []PredicateFailureReason{ErrPodNotMatchHostName}, nil
 }
 
-// PodFitsHostPorts is a wrapper around PodFitsHostPortsPredicate. This is needed until
+// podFitsHostPorts is a wrapper around PodFitsHostPortsPredicate. This is needed until
 // we are able to get rid of the FitPredicate function signature.
-// TODO(#85822): remove this function once predicate registration logic is deleted.
-func PodFitsHostPorts(pod *v1.Pod, _ Metadata, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
-	return PodFitsHostPortsPredicate(pod, nil, nodeInfo)
-}
-
-// PodFitsHostPortsPredicate checks if a node has free ports for the requested pod ports.
-func PodFitsHostPortsPredicate(pod *v1.Pod, meta []*v1.ContainerPort, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
-	wantPorts := meta
-	if wantPorts == nil {
-		// Fallback to computing it.
-		wantPorts = schedutil.GetContainerPorts(pod)
-	}
-	if len(wantPorts) == 0 {
-		return true, nil, nil
-	}
-
-	existingPorts := nodeInfo.UsedPorts()
-
+func podFitsHostPorts(pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
+	wantPorts := schedutil.GetContainerPorts(pod)
 	// try to see whether existingPorts and  wantPorts will conflict or not
+	existingPorts := nodeInfo.UsedPorts()
 	for _, cp := range wantPorts {
 		if existingPorts.CheckConflict(cp.HostIP, string(cp.Protocol), cp.HostPort) {
 			return false, []PredicateFailureReason{ErrPodNotFitsHostPorts}, nil
 		}
 	}
-
 	return true, nil, nil
 }
 
 // GeneralPredicates checks a group of predicates that the kubelet cares about.
 // DEPRECATED: this exist only because kubelet uses it. We should change kubelet to execute the individual predicates it requires.
-func GeneralPredicates(pod *v1.Pod, meta Metadata, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
+func GeneralPredicates(pod *v1.Pod, _ interface{}, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
```

Comment:
Just a question, this would be move into kubelet?

Reply:
may be, not sure yet.